fili-core/src/test/groovy/com/yahoo/bard/webservice/data/config/table/DimensionListPartitionTableDefinitionSpec.groovy

```diff
@@ -0,0 +1,98 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.data.config.table
+
+import static com.yahoo.bard.webservice.data.time.ZonedTimeGrainSpec.DAY_UTC
+
+import com.yahoo.bard.webservice.data.config.ResourceDictionaries
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.data.dimension.Dimension
+import com.yahoo.bard.webservice.data.dimension.DimensionDictionary
+import com.yahoo.bard.webservice.data.dimension.DimensionField
+import com.yahoo.bard.webservice.metadata.DataSourceMetadataService
+import com.yahoo.bard.webservice.table.PartitionCompositeTable
+import com.yahoo.bard.webservice.table.PhysicalTable
+import com.yahoo.bard.webservice.table.PhysicalTableSchema
+import com.yahoo.bard.webservice.table.availability.Availability
+import com.yahoo.bard.webservice.table.resolver.DimensionIdFilter
+
+import spock.lang.Specification
+
+class DimensionListPartitionTableDefinitionSpec extends Specification {
+
+    DimensionField keyField = Mock(DimensionField)
+
+    String dimensionName = "testDimension"
+
+    Dimension testDimension = Mock(Dimension) {
+        getApiName() >> dimensionName
+        getKey() >> keyField
+    }
+
+    DimensionDictionary dimensionDictionary = new DimensionDictionary()
+    ResourceDictionaries resourceDictionaries = new ResourceDictionaries()
+
+    TableName part1Name = TableName.of("part1")
+    TableName part2Name = TableName.of("part2")
+
+    TableName dataSource1 = TableName.of("part1_ds1")
+    TableName dataSource2 = TableName.of("part1_ds2")
+    TableName dataSource3 = TableName.of("part2_ds1")
```

Comment:
Since these aren't java identifiers I think readability is the biggest concern and this way is clearly more readable.